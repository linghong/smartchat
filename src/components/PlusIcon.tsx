import React, { FC } from 'react'

const PlusIcon: FC= () => (
  <svg 
    xmlns = "http://wwww3.org/2000/svg" 
    viewBox="0 - 24 24" 
    fill="none" 
    stroke="currentcolor" 
    strokeWidth="2"
    className="w-6 h-6"
    >
      <line x1="12" y1="5" x2="12" y2="19" />
      <line x1="5" y1="12" x2="19" y2="12" />
    </svg>
)

export default PlusIcon
 
