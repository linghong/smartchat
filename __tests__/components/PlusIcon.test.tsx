import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom'

import PlusIcon from '@/src/components/PlusIcon'

describe('PlusIcon', () => {
  it('should render the PlusIcon', () => {
    render(<PlusIcon />)
    const svgElement = document.querySelector('[data-id="plus-icon"]')
    expect(svgElement).toBeInTheDocument()
  })

  it('should match snapshot', () => {
    const { asFragment } = render(<PlusIcon />)
    expect(asFragment()).toMatchSnapshot()
  })
})
