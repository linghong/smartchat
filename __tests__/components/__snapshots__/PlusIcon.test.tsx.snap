// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlusIcon should match snapshot 1`] = `
<DocumentFragment>
  <svg
    class="w-6 h-6"
    data-id="plus-icon"
    fill="none"
    stroke="currentcolor"
    stroke-width="2"
    viewBox="0 0 24 24"
    xmlns="http://wwww3.org/2000/svg"
  >
    <line
      x1="12"
      x2="12"
      y1="5"
      y2="19"
    />
    <line
      x1="5"
      x2="19"
      y1="12"
      y2="12"
    />
  </svg>
</DocumentFragment>
`;
